apps/web/modules/api/management/responses/types/responses.ts

```diff
@@ -0,0 +1,55 @@
+import { Contact, Response, ResponseNote, Tag, User } from "@prisma/client";
+import { z } from "zod";
+import { ZResponse } from "@formbricks/database/zod/responses";
+
+export type TResponseNew = Response & {
+  contact?: Pick<Contact, "id" | "userId">;
+  notes?: (Pick<ResponseNote, "id" | "text" | "createdAt" | "updatedAt" | "isEdited" | "isResolved"> & {
+    user: Pick<User, "id" | "name">;
+  })[];
+  tags: Tag[];
+};
+
+export const ZGetResponsesFilter = z.object({
+  surveyId: z.string().cuid2().optional(),
+  limit: z.coerce.number().default(10).optional(),
+  skip: z.coerce.number().default(0).optional(),
+  sortBy: z.enum(["createdAt", "updatedAt"]).default("createdAt").optional(),
+  order: z.enum(["asc", "desc"]).default("asc").optional(),
+  startDate: z.coerce.date().optional(),
+  endDate: z.coerce.date().optional(),
+});
+
+export type TGetResponsesFilter = z.infer<typeof ZGetResponsesFilter>;
+
+export const ZResponseInput = ZResponse.pick({
```

Comment:
Same for the response input: We should only allow the fields that also prisma allows to create the response object in the database without any other fields.
So for example the `userId` should not be part of it.